### PR TITLE
[SPARK-38490][SQL][INFRA] Add Github action test job for ANSI SQL mode

### PR DESCRIPTION
**Pull request by gengliangwang**
### What changes were proposed in this pull request?

Add Github action test job for ANSI SQL mode. It will be triggered after each commit push on the master branch of apache/spark.
image

To make the implementation easy, ANSI SQL mode becomes enabled if ENV variable `SPARK_ANSI_SQL_MODE` is set as `true`

### Why are the changes needed?

Testing the ANSI SQL mode on each commit push, so that we can find issues in time.

### Does this PR introduce _any_ user-facing change?

Yes, ANSI SQL mode becomes enabled if ENV variable `SPARK_ANSI_SQL_MODE` is set as `true`. This won't be in documentation since it is for testing purpose.

### How was this patch tested?

Manually try on my repo: https://github.com/gengliangwang/spark/actions/runs/1961176426

### CODE REVIEW

File: sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala [hunk 1]
```diff
@@ -2706,7 +2706,7 @@ object SQLConf {
       "standard directly, but their behaviors align with ANSI SQL's style")
     .version("3.0.0")
     .booleanConf
-    .createWithDefault(false)
+    .createWithDefault(sys.env.get("SPARK_ANSI_SQL_MODE").contains("true"))
```
Comment (cloud-fan) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala:2709]: will `sys.env.get("SPARK_ANSI_SQL_MODE")` be null?

Comment (gengliangwang) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala:2709]: The result type of `sys.env.get("SPARK_ANSI_SQL_MODE")` is Option[String]. It is totally fine here.

File: .github/workflows/ansi_sql_mode_test.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+name: ANSI SQL mode test
+
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  ansi_sql_test:
+    uses: ./.github/workflows/build_and_test.yml
```
Comment (HyukjinKwon) [inline .github/workflows/ansi_sql_mode_test.yml:29]: oh wow. This is nice. Now we can reuse workflow files. cc @dongjoon-hyun FYI

File: .github/workflows/ansi_sql_mode_test.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    if: github.repository == 'apache/spark'
```
Comment (HyukjinKwon) [inline .github/workflows/ansi_sql_mode_test.yml:30]: Should we add it as a scheduled job though? Running it for every commit might pressure the GA resource a bit intentively.

File: .github/workflows/ansi_sql_mode_test.yml (hunk 2 above)
Comment (gengliangwang) [reply .github/workflows/ansi_sql_mode_test.yml:29]: Yes. I suggest that we move all the scheduled workflows to separated files.

Comment (HyukjinKwon) [reply .github/workflows/ansi_sql_mode_test.yml:29]: Yeah, we should separate them. This is nice. @Yikun too FYI

Comment (dongjoon-hyun) [reply .github/workflows/ansi_sql_mode_test.yml:29]: Ya, it looks like a nice tip. Thanks, @HyukjinKwon .

Comment (Yikun) [reply .github/workflows/ansi_sql_mode_test.yml:29]: Cool, really good, also is an inspiration to reuse workflow in downstream repo.

Looks like there are some limitations we can only separated our workflow which is not using `strategy`.

[1] https://docs.github.com/en/actions/using-workflows/reusing-workflows#limitations

Comment (gengliangwang) [reply .github/workflows/ansi_sql_mode_test.yml:29]: @Yikun the limitation is for the caller. The strategy in "build_and_test.yml" still works.

**Comment by gengliangwang** [comment]
This one is ready. The test result https://github.com/gengliangwang/spark/runs/5491483010 is the same as changing the SQLConf manually: https://github.com/apache/spark/pull/34970

**Review by cloud-fan — APPROVED** [review]

**Pull request closed without merging**